lib/matplotlib/widgets.py

```diff
@@ -1097,15 +1188,25 @@ def set_active(self, index):
         if index not in range(len(self.labels)):
             raise ValueError(f'Invalid CheckButton index: {index}')
 
-        cross_facecolors = self._crosses.get_facecolor()
-        cross_facecolors[index] = colors.to_rgba(
-            "black"
+        check_facecolors = self._checks.get_facecolor()
+        check_facecolors[index] = colors.to_rgba(
+            self._check_facecolors[index]
             if colors.same_color(
-                cross_facecolors[index], colors.to_rgba("none")
+                check_facecolors[index], colors.to_rgba("none")
             )
             else "none"
         )
-        self._crosses.set_facecolor(cross_facecolors)
+        self._checks.set_facecolor(check_facecolors)
+
+        check_edgecolors = self._checks.get_edgecolor()
+        check_edgecolors[index] = colors.to_rgba(
+            self._check_edgecolors[index]
+            if colors.same_color(
+                check_edgecolors[index], colors.to_rgba("none")
```

Comment:
Worth hard coding whatever "none" is in RGBA to improve performance a little here?